### ISSUE TITLE
New ways to pull in compose files, with versioning

### ISSUE THREAD

**Issue by 486** [francescou/docker-compose-ui#22]
Hi,

I like your project very much! My company is looking for ways to make it very easy to have access to all our docker compositions. Currently, our docker-compose.yml files plus additional stuff are in multiple git repositories. There would be tagged commits to mark different release versions. 

Now it would be great if our compositions would show up in docker-compose-ui (let's call it dcu), with the option to build / pull / start / etc. any of the tagged versions. I can think of multiple ways to archive this:
- dcu could be extended to read from different "sources", i.e. git, SVN, etc. as an additional way besides the directory
- a separate container / service could provision the dcu directory with the docker-compose.yml stuff (clunky for versioning)
- dcu could implement an "open docker compose registry API" where the registry could be configured. I see that you already have created something like that, but as far as I understand your registry is closed source (I could re-implement it)

In any case, there would have to be a solution for choosing between multiple release versions of a composition (similar to Docker Hub). 

What do you think about this? If we could agree on a design, I would try to support you with the development (I am mostly a node.js developer, but your code style is super clean so it shouldn't be a problem).

**Comment by francescou** [comment]
Hello, 

thank you for your suggestion.
I am interested in exploring a solution like this, I would like to ask you some question about your current workflow. 
If you want add me on skype francesco.uliana and we can discuss about this.

Thanks

**Comment by francescou** [comment]
I'm working on a prototype to manage docker-compose projects from a git repository, see branch _gitrepo_ or run:

```
docker run -p 5000:5000 -e GIT_REPO=https://github.com/francescou/docker-continuous-deployment.git -v /var/run/docker.sock:/var/run/docker.sock  francescou/docker-compose-ui:gitrepo
```

**Comment by 486** [comment]
That sounds amazing! I didn't get around to contact you, sorry for that. I will try that branch ASAP.

**Comment by WTFKr0** [comment]
Im interested in this too :+1:

**Comment by francescou** [comment]
Hi @486 and @WTFKr0 

I've updated the [gitrepo](https://github.com/francescou/docker-compose-ui/tree/gitrepo) branch 

tested against github and a private gitlab instance repositories, it seems to work fine, e.g.

```
docker run --rm --name docker-compose-ui -p 5000:5000 \
  -v /var/run/docker.sock:/var/run/docker.sock  \
  -e GIT_REPO=http://username:password@my.gitlab/devops/docker-compose-dev.git \
  francescou/docker-compose-ui:gitrepo
```

feedback is welcome!

**Comment by 486** [comment]
Hi, you are doing amazing work and I still didn't get around to check it out which is an absolute shame. I will report back very soon, promise.

**Comment by francescou** [comment]
released in [0.18](https://github.com/francescou/docker-compose-ui/releases/tag/v0.18), [instructions](https://github.com/francescou/docker-compose-ui#load-projects-from-a-git-repository-experimental):

```
docker run \
--name docker-compose-ui \
-p 5000:5000 \
-v /var/run/docker.sock:/var/run/docker.sock  \
-e GIT_REPO=https://github.com/francescou/docker-compose-ui.git \
francescou/docker-compose-ui:0.18
```
